### ISSUE TITLE
<link> elements in a document <body> do not trigger errors

### ISSUE THREAD

**Issue by tmcgill** [validator/validator#532]
[HTML 5/5.1 recommendations](https://www.w3.org/TR/html51/document-metadata.html#the-link-element) indicate that the `link` element is restricted to the `head` element. When run through the Nu validator (version 17.6.25), markup with `link` elements in the `body` do not properly trigger errors.

**Comment by unor** [comment]
Can you give an example?

The below test document gives the error: "A `link` element must not appear as a descendant of a `body` element unless […]"

```
<!DOCTYPE html>
<html>
<head>
<title>Test</title>
</head>
<body>
<link rel="canonical" href="/">
</body>
</html>
```

**Comment by tmcgill** [comment]
Using Nu instances at https://validator.w3.org/ and https://checker.html5.org/...

If I use your example, but change `rel="canonical"` to `rel="stylesheet"`, no error is thrown.

**Comment by unor** [comment]
See [body-ok](https://www.w3.org/TR/2017/WD-html52-20170509/links.html#body-ok) (HTML 5.2 WD):

> Keywords that are **body-ok** affect whether `link` elements are allowed in the body.

`stylesheet` is body-ok.

**Comment by tmcgill** [comment]
Ok, I assumed Nu validated against W3C recommendations, not working drafts.

**Comment by sideshowbarker** [comment]
Closing this out per https://github.com/validator/validator/issues/532#issuecomment-312167459. But feel free to comment here further if any additional clarification is needed.
